File: go/fixture/go-mini/internal/models/tags.go
package models

import (
	"slices"
	"strings"
)

func NormalizeTags(tags []string) []string {
	var out []string
	for _, t := range tags {
		t = strings.TrimSpace(t)
		if t == "" || slices.Contains(out, t) {
			continue
		}
		if strings.HasPrefix(t, "region:") && !isRegionCode(t[7:]) {
			continue
		}
		out = append(out, t)
	}
	return out
}

func isRegionCode(code string) bool {
	return code == "eu" || code == "us" || code == "ap"
}

File: py/fixture/py-mini/internal/models/tags.py
"""Tag lists as devices report them."""


def normalize_tags(tags: list[str]) -> list[str]:  # noqa: D103
    out: list[str] = []
    for raw in tags:
        t = raw.strip()
        if not t or t in out:
            continue
        if t.startswith("region:") and not _is_region_code(t[7:]):
            continue
        out.append(t)
    return out


def _is_region_code(code: str) -> bool:
    return code == "eu" or code == "us" or code == "ap"  # noqa: PLR1714, PLR2004  # TODO
